### PR TITLE
Fixes #24

### LINKED ISSUE #24: luamin from bash script not working
Hi!

I tried to use luamin in combination with git client hooks, when i ran into an error.

The script looks like this:

        #!/bin/sh
        #
        
        cat file1 file2 file3 > concat.lua
        mkdir -p dist
        luamin -f  concat.lua > minified.lua
        rm concat.lua
        git add dist

After running the hook, the content of `minified.lua` is.
        Error: no such file. (``)

`concat.lua` looks perfectly fine. If I run `luamin -f concat.lua > minified.lua` from commandline i don't get errors as well and the minified file is created as planned.

Bash seems to add a magical empty element to the end of the arguments.
This error should be fixed by checking, if snippet is an empty string in bin/luamin: 55
If i find some time, I'll test it on my own and create a pull request.

greetings

Nuke

### PR DESCRIPTION
**Pull request by Nukesor**
Hi!

I can't reopen Issue #24 , but i created a working test case on a branch in our project.
[Minify Test Branch](https://github.com/lovetoys/lovetoys/tree/minify)
There is no double space in my original hook script. Maybe you can take a look at it and see it by yourself.
Run `make test` to commit and reset, to trigger the hook.
This simple update fixes the problem.

### CODE REVIEW

File: bin/luamin [hunk 1]
```diff
@@ -53,41 +53,43 @@
...
+                    log(
+                        '\nStack trace using luamin@%s and luaparse@%s:\n',
+                        luamin.version,
+                        require('luaparse').version
+                    );
+                    log(error.stack);
+                    return process.exit(1);
+                }
+            }
+        });
```
Comment (mathiasbynens) [inline bin/luamin]: Please respect the existing coding style w.r.t. indentation.

**Comment by Nukesor** [comment]
Sry, i didn't notice, that you're using tabs. Should be fine now.

**Comment by raffomania** [comment]
Is there a chance to get this merged? We're really depending on script integration, it would be awesome to see this fix in upstream!

**Comment by mathiasbynens** [comment]
Could you provide a reduced test case as a gist? Thanks.

**Comment by raffomania** [comment]
Hi @mathiasbynens,

Try placing https://gist.github.com/raffomania/42066ad7afaabfef5c75 in a pre-commit git hook (`.git/hooks/pre-commit`). The real problem here is git executing the hook without a TTY context, causing the luamin script to expect input from stdin.

Let me know if this helps.

**Comment by mathiasbynens** [comment]
> The real problem here is git executing the hook without a TTY context, causing the luamin script to expect input from stdin.

Then it seems to me we should patch *that* rather than hiding the problem, like this patch does.

**Comment by Nukesor** [comment]
You're totally right. I'll close this pull request.

**Pull request closed without merging**